File: app/Model/OrganizationUser.js
'use strict';

const Lucid = use('Lucid');

class OrganizationUser extends Lucid {
	user() {
		return this.hasOne('App/Model/User');
	}

	organization() {
		return this.hasOne('App/Model/Organization');
	}
}

module.exports = OrganizationUser;

File: web/models/organization_user.ex
defmodule Poker.OrganizationUser do
  use Poker.Web, :model

  schema "organizations_users" do
    field :role, :string
    belongs_to :organization, Poker.Organization
    belongs_to :user, Poker.User

    timestamps()
  end

  # Changesets

  def changeset(struct, _params \\ %{}) do
    struct
    |> validate_required([:role])
    |> validate_inclusion(:role, ["owner", "member"])
  end

  def create_changeset(struct, params \\ %{}) do
    struct
    |> cast(params, [:role, :user_id, :organization_id])
    |> changeset(params)
  end
end
